table/server/common/src/main/java/alluxio/table/common/udb/UdbMountSpec.java

```diff
@@ -0,0 +1,589 @@
+      } else {
+        return !mExcluded.has(name);
+      }
+    }
+  }
+
+  static final class TablePartitionWrapper
+      extends InclusionExclusionWrapper<TablePartitionNamePatternWrapper> {
+    TablePartitionWrapper(
+        TablePartitionNamePatternWrapper included,
+        SimpleNamePatternWrapper excluded) {
+      super(included, excluded);
+    }
+
+    boolean hasTable(String tableName) {
+      return has(tableName);
+    }
+
+    boolean hasFullTable(String tableName) {
+      if (mExcluded.isEmpty()) {
+        return mIncluded.hasFullTable(tableName);
+      } else {
+        // tables that are implicitly included by being NOT excluded are meant to be fully bypassed
+        // since there is no way to specify partitions in an exclusion list
+        return !mExcluded.has(tableName);
+      }
+    }
+
+    boolean hasPartition(String tableName, String partName) {
+      if (mExcluded.isEmpty()) {
+        return mIncluded.hasPartition(tableName, partName);
+      } else {
+        return !mExcluded.has(tableName);
+      }
+    }
+  }
+
+  /**
+   * Type alias for {@link InclusionExclusionWrapper<SimpleNamePatternWrapper>}.
+   */
+  static final class SimpleWrapper extends InclusionExclusionWrapper<SimpleNamePatternWrapper> {
+    SimpleWrapper(SimpleNamePatternWrapper included, SimpleNamePatternWrapper excluded) {
+      super(included, excluded);
+    }
+  }
+
+  /**
+   * Wrapper for table and partition specs, only appears in bypass-include list.
+   */
+  static class TablePartitionNamePatternWrapper extends NamePatternWrapper {
+    private final ImmutableMap<String, Set<SimpleWrapper>> mTablePartMap;
+
+    TablePartitionNamePatternWrapper(Set<String> fullTableNames, Set<Pattern> patterns) {
+      this(fullTableNames, patterns, Collections.emptyMap());
+    }
+
+    TablePartitionNamePatternWrapper(Set<String> fullTableNames, Set<Pattern> patterns,
+                                     Map<String, Set<SimpleWrapper>> tablePartMap) {
+      super(fullTableNames, patterns);
+      mTablePartMap = ImmutableMap.copyOf(tablePartMap);
+    }
+
+    static TablePartitionNamePatternWrapper empty() {
+      return new TablePartitionNamePatternWrapper(Collections.emptySet(), Collections.emptySet());
+    }
+
+    @Override
+    boolean isEmpty() {
+      return super.isEmpty() && mTablePartMap.isEmpty();
+    }
+
+    boolean hasFullTable(String tableName) {
+      // if a table is listed with an explicit literal name, then the partition specification
+      // or regex patterns should not be consulted:
+      // here use parent's hasExplicit since self's takes into account partition specs
+      if (super.hasExplicit(tableName)) {
+        return true;
+      }
+      if (mTablePartMap.containsKey(tableName)) {
+        // otherwise, check if it has an empty partition specification,
+        // or a non-empty partition spec.
+        // either way, it is explicitly listed, so no need to consult the regex patterns.
```

Comment:
By these words, I mean in the following example:

```
{
  "tables": [
    {"regex": "table\\d"},
    {"table": "table1", "partitions": ["part1"]}
  ]
}
```

`table1` has non-empty partition specifications, although it is also captured by the regex, its name is explicitly listed through table-partition specification, so we consider it not a fully bypassed table.